include/qfm/yahoo_finance_api.hpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2023 David Alvarez Rosa, Matteo Durante
+ */
+
+#pragma once
+
+#include "qfm/asset/asset_ticker.hpp"
+#include "qfm/finance_api.hpp"
+
+namespace qfm {
+
+class YahooFinanceApi : public FinanceApi {
+ public:
+  YahooFinanceApi() = default;
+  double QueryBidPrice(
+      const asset::AssetTicker& asset_ticker) const noexcept override;
+  double QueryAskPrice(
+      const asset::AssetTicker& asset_ticker) const noexcept override;
```

Comment:
I think that it would be better if we had a `QueryQuote` method returning the pair as well, so that in order to get both values we don't need to call both methods, which perform exactly the same operation. Similarly, `MarketDataProvider` should have a `GetAssetQuote` method as well.